### PR TITLE
GO1P-26987 Updating to get Dimensions for the GET 

### PR DESCRIPTION
**Pull request by Go1Nick**

### CODE REVIEW

File: dimensions/DimensionRepository.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class DimensionRepository
+{
+    private $db;
+
+    public function __construct(Connection $db) {
+        $this->db = $db;
+    }
+
+    public static function install(Schema $schema)
```
Comment (andytruong) [inline dimensions/DimensionRepository.php]: I don't this this should be role of repository.

File: dimensions/DimensionRepository.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        if (!$schema->hasTable('dimensions')) {
+            $table = $schema->createTable('dimensions');
+            $table->addColumn('id', TYPE::INTEGER, ['autoincrement' => true]);
```
Comment (andytruong) [inline dimensions/DimensionRepository.php]: unsigned?

File: dimensions/DimensionRepository.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            $table->setPrimaryKey(['id'], 'pk_id');
+            $table->addColumn('parent_id', TYPE::INTEGER, ['notnull' => false]);
+            $table->addColumn('name', TYPE::STRING);
+            $table->addColumn('type', TYPE::STRING);
+            $table->addColumn('created_date', TYPE::DATETIME);
+            $table->addColumn('modified_date', TYPE::DATETIME);
```
Comment (andytruong) [inline dimensions/DimensionRepository.php]: no index needed?

File: dimensions/DimensionRepository.php (hunk 1 above)
Comment (Go1Nick) [reply dimensions/DimensionRepository.php]: Ok, what would you recommend it be?

File: lo/LoHelper.php [hunk 4]
```diff
@@ -203,7 +204,14 @@ private static function getAttributes(Connection $db, array $ids)
                         $newVal = $attribute->value;
                         $attribute->value = [];
                         foreach ($newVal as $val) {
-                            $attribute->value[strval($val)] = "";
+                            if ($attribute->dimension_id) {
+                                $dimensions = DimensionHelper::loadAllForType($db, $attribute->dimension_id);
```
Comment (andytruong) [inline lo/LoHelper.php]: We can optimise this by load all dimensions using one query.

File: dimensions/DimensionRepository.php (hunk 2 above)
Comment (Go1Nick) [reply dimensions/DimensionRepository.php]: I just took this directly from the Dimensions microservice

File: lo/LoHelper.php [hunk 5]
```diff
@@ -576,4 +584,30 @@ public static function allowReuseEnrolment(stdClass $lo): bool
...
+        }
+        if ($lookup->isArray) {
+            $value = json_decode($value);
+        }
+        if ($lookup->attributeType === LoAttributeTypes::DIMENSION) {
+            $newVal = $value;
+            $value = [];
+            foreach ($newVal as $val) {
+                if (isset($lookup->dimensionId)) {
+                    $dimensions = DimensionHelper::loadAllForType($db, $lookup->dimensionId);
```
Comment (andytruong) [inline lo/LoHelper.php]: Should avoid multiple queries.

File: lo/LoHelper.php (hunk 4 above)
Comment (Go1Nick) [reply lo/LoHelper.php]: This is using the https://github.com/go1com/util_core/blob/master/dimensions/DimensionHelper.php

File: tests/lo/LoHelperTest.php [hunk 6]
```diff
@@ -579,8 +579,16 @@ public function testPremiumFlag()
...
         $this->createAttributeLookup($this->go1, LoAttributes::REGION_RESTRICTIONS, LoAttributes::machineName(LoAttributes::REGION_RESTRICTIONS), 'DIMENSION', 'video',
-            '["ALWAYS", "FOR_PUBLISH"]', '["Author"]', null, 1);
-
+            '["ALWAYS", "FOR_PUBLISH"]', '["Author"]', null, 1, 2);
+
+        $this->go1->insert('dimensions', [
+            'id'             => 3,
+            'parent_id'      => 0,
+            'name'           => "NAME",
+            'type'           => "2",
```
Comment (andytruong) [inline tests/lo/LoHelperTest.php]: Can we have some self-describeing constant instead of 2?

File: lo/LoHelper.php (hunk 5 above)
Comment (Go1Nick) [reply lo/LoHelper.php]: This is now the only usage, I have updated the above to call this function

File: dimensions/DimensionRepository.php (hunk 3 above)
Comment (Go1Nick) [reply dimensions/DimensionRepository.php]: None was needed when originally when creating the dimensions table so I assume its not needed here

Comment (andytruong) [reply dimensions/DimensionRepository.php]: At least, you please provide some comment.

Comment (pvmchau) [reply dimensions/DimensionRepository.php]: The main queried columns or set of columns should be indexed to improve the searching performance in logic.

Comment (Go1Nick) [reply dimensions/DimensionRepository.php]: I have added index to the two main used columns, type and name

**Pull request merged**